service/resource/vaultpki/current.go

```diff
@@ -0,0 +1,47 @@
+package vaultpki
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/vaultpki"
+
+	"github.com/giantswarm/cert-operator/service/key"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var vaultPKIState VaultPKIState
+
+	{
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for the Vault PKI in the Vault API")
+
+		vaultPKIState.Backend, err = r.vaultPKI.GetBackend(key.ClusterID(customObject))
+		if vaultpki.IsNotFound(err) {
+			r.logger.Log("cluster", key.ClusterID(customObject), "debug", "did not find the Vault PKI in the Vault API")
```

Comment:
We shouldn't rely on the fact that `GetBackend` returns `nil` backend when error is not `nil`

Reply:
Done.